src/functions-and-arrays.js

```diff
@@ -78,37 +210,81 @@ const wordsCount = [
-  [21, 36, 23, 9, 75, 0, 76, 44, 20, 45, 35, 14, 0, 61, 33, 97, 34, 31, 33, 95],
-  [78, 17, 53, 28, 22, 75, 31, 67, 15, 94, 3, 80, 4, 62, 16, 14, 9, 53, 56, 92],
-  [16, 39, 5, 42, 96, 35, 31, 47, 55, 58, 88, 24, 0, 17, 54, 24, 36, 29, 85, 57],
-  [86, 56, 0, 48, 35, 71, 89, 7, 5, 44, 44, 37, 44, 60, 21, 58, 51, 54, 17, 58],
-  [19, 80, 81, 68, 5, 94, 47, 69, 28, 73, 92, 13, 86, 52, 17, 77, 4, 89, 55, 40],
-  [4, 52, 8, 83, 97, 35, 99, 16, 7, 97, 57, 32, 16, 26, 26, 79, 33, 27, 98, 66],
-  [88, 36, 68, 87, 57, 62, 20, 72, 3, 46, 33, 67, 46, 55, 12, 32, 63, 93, 53, 69],
-  [4, 42, 16, 73, 38, 25, 39, 11, 24, 94, 72, 18, 8, 46, 29, 32, 40, 62, 76, 36],
-  [20, 69, 36, 41, 72, 30, 23, 88, 34, 62, 99, 69, 82, 67, 59, 85, 74, 4, 36, 16],
-  [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
-  [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
-];
-
-function greatestProduct() {}
+function howManyTimes(words, word) {
+  let counter = 0;
+
+  if (words.length === 0) {
+    return 0;
+  }
+  
+  for (let i = 0; i < words.length; i++) {
+    if (words[i] === word) {
+      counter++;
+    }
+  }
+  return counter;  
+}
 
+// Bonus - Iteration #8: Product of adjacent numbers
+
+function greatestProduct(matrix) {
+  let productRow;
+  let productColumn;
+  let maxProduct = 0;
+
+  // Calculates greatest product of any rows
+  for (let i = 0; i < matrix.length; i++) {
+    for (let j = 0; j < matrix.length - 3; j++) {
+      productRow = matrix[i][j] * matrix[i][j+1] * matrix[i][j+2] * matrix[i][j+3];
+      if (productRow > maxProduct) {
+        maxProduct = productRow;
+      }
+    }
+  }
+    
+  // Calculates greatest product of any columns
+  for (let i = 0; i < matrix.length - 3; i++) {
+    for (let j = 0; j < matrix.length; j++) {
+      productColumn = matrix[i][j] * matrix[i+1][j] * matrix[i+2][j] * matrix[i+3][j];
+      if (productColumn > maxProduct) {
+        maxProduct = productColumn;
+      }
+    }
+  }
+    
+  return maxProduct;
+}
 
+// Bonus - Iteration #8.1: Product of diagonals
+
+function greatestProductDiagonal(matrix) {
+  let product;
+  let maxProduct = 0;
+
+  // Calculates greatest product going to the down/right
+  for (let i = 0; i < matrix.length - 3; i++) {
+    for (let j = 0; j < matrix.length - 3; j++) {
+      if (i + 3 < matrix.length && j + 3 < matrix.length) {
+        product = matrix[i][j] * matrix[i+1][j+1] * matrix[i+2][j+2] * matrix[i+3][j+3];
+        if (product > maxProduct) {
+          maxProduct = product;
+        }
+      }
+    }
+  }
+    
+  // Calculates greatest product going to the down/left
+  for (let i = 0; i < matrix.length - 3; i++) {
+    for (let j = 3; j < matrix.length; j++) {
+      product = matrix[i][j] * matrix[i+1][j-1] * matrix[i+2][j-2] * matrix[i+3][j-3];
+      if (product > maxProduct) {
+        maxProduct = product;
+      }
+    }
+  }
+    
+  return maxProduct;
+}
```

Comment:
Maybe you can try putting this one in only one nested loop too? If you don't manage we can send you the solution :). 

Really well done!!!